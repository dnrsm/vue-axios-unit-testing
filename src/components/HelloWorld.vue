<template>
  <div class="hello">
    <ul>
      <li v-for="(user, index) in filteredUsers" :key="index">
        {{ user.id }}: {{ user.name }}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  name: "HelloWorld",
  data() {
    return {
      users: []
    };
  },
  computed: {
    filteredUsers() {
      if (Object.keys(this.users).length === 0) return;
      return this.users.slice(0, 3);
    }
  },
  created() {
    this.request();
  },
  methods: {
    request() {
      this.$axios
        .get("https://jsonplaceholder.typicode.com/users")
        .then(res => {
          this.users = res.data;
        });
    }
  }
};
</script>
