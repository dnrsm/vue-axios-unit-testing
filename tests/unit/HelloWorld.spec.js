import { shallowMount } from "@vue/test-utils";
import HelloWorld from "@/components/HelloWorld.vue";
import mockAxios from "axios";

const response = {
  data: [
    {
      id: 1,
      name: "Leanne Graham"
    },
    {
      id: 2,
      name: "Ervin Howell"
    },
    {
      id: 3,
      name: "Clementine Bauch"
    }
  ]
};

describe("HelloWorld.vue", () => {
  mockAxios.get.mockReturnValue(Promise.resolve(response));
  const wrapper = shallowMount(HelloWorld, {
    mocks: {
      $axios: mockAxios
    }
  });
  it("正常にレンダリングされる", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.exists()).toBe(true);
  });
  it("一度だけ呼ばれる", () => {
    expect(mockAxios.get).toHaveBeenCalledTimes(1);
  });
  it("usersのエンドポイントが引数として与えられる", () => {
    expect(mockAxios.get).toBeCalledWith(
      "https://jsonplaceholder.typicode.com/users"
    );
  });
  it("usersの一番目のname ⇛ Leanne Graham", () => {
    expect(wrapper.vm.users[0].name).toMatch("Leanne Graham");
  });
});
