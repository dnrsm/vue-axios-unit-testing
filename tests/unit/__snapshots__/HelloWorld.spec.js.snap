// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue 正常にレンダリングされる 1`] = `
<div class="hello">
  <ul>
    <li>
      1: Leanne Graham
    </li>
    <li>
      2: Ervin Howell
    </li>
    <li>
      3: Clementine Bauch
    </li>
  </ul>
</div>
`;
